src/gsy_e/models/market/forward.py

```diff
@@ -0,0 +1,138 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from abc import abstractmethod
+from calendar import monthrange
+from typing import Optional, TYPE_CHECKING
+
+from gsy_framework.constants_limits import ConstSettings
+from pendulum import DateTime, duration
+
+from gsy_e.gsy_e_core.blockchain_interface import NonBlockchainInterface
+from gsy_e.models.market import GridFee
+from gsy_e.models.market.future import FutureMarkets
+
+if TYPE_CHECKING:
+    from gsy_e.models.config import SimulationConfig
+    from gsy_e.models.area.event_dispatcher import AreaDispatcher
+
+
+class ForwardMarketBase(FutureMarkets):
+    """Base class for forward markets"""
+
+    def __init__(self, bc: Optional[NonBlockchainInterface] = None,
+                 notification_listener: Optional["AreaDispatcher"] = None,
+                 readonly: bool = False,
+                 grid_fee_type: int = ConstSettings.MASettings.GRID_FEE_TYPE,
+                 grid_fees: Optional[GridFee] = None,
+                 name: Optional[str] = None) -> None:
+        # pylint: disable=too-many-arguments
+        if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
+            return
+
+        super().__init__(bc=bc, notification_listener=notification_listener,
+                         readonly=readonly, grid_fee_type=grid_fee_type,
+                         grid_fees=grid_fees, name=name)
+
+    @staticmethod
+    @abstractmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        """Return time when the market block starts."""
+
+    @staticmethod
+    @abstractmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        """Return time when the market block ends."""
+
+    @staticmethod
+    @abstractmethod
+    def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
+        """Return duration of market slots inside the market block."""
+
+    def create_future_market_slots(self, current_market_time_slot: DateTime,
+                                   config: "SimulationConfig") -> None:
+        if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
+            return
+        self._create_future_market_slots(self._get_start_time(current_market_time_slot),
+                                         self._get_end_time(current_market_time_slot), config)
+
+
+class DayForwardMarket(ForwardMarketBase):
+    """Day forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(days=1)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(weeks=1, days=1, hours=-1)
```

Comment:
Change the order of `.set` and `.add` as discussed below in `YearForwardMarket`.
Also I believe the end time should be like this:
```python
return current_time.add(days=2).set(hour=0, minute=0)
````

Reply:
For the latter it looks correct IMO. Again copying from the documentation:

`For hourly products, the market opens one week in advance and closes one hour in advance.`